test/core/CoreTest.cpp

```diff
@@ -42,5 +56,71 @@ TEST_F(CoreTest, changeProductQuantityShouldReturnSuccess)
     EXPECT_EQ(core.changeProductQuantity("decrease",0), Core::MethodResult::SUCCESS);
 }
 
+TEST_F(CoreTestProduct, createProductShouldReturnSuccess)
+{
+    EXPECT_EQ(core.createProduct(product), Core::MethodResult::SUCCESS);
+}
+
+TEST_F(CoreTestProduct, updateProductShouldReturnSuccess)

```

Comment:
Jak masz ochotę, możesz spróbwać wejśc na wyższy poziom, i użyć tutaj testów parametrycznych. 
https://www.sandordargo.com/blog/2019/04/24/parameterized-testing-with-gtest

```
struct ProductTestStruct
{
int id;
Core::MethodResult operationResutl;
};

class CoreTestProductParametrized : public ::testing::TestWithParam<ProductTestStruct>
                                                           , public CoreTestProduct
{};

TEST_P(CoreTestProductParametrized, UpdateOperation) {
    const auto id year = GetParam().id;
    const auto result = GetParam().operationResutl;
    
    EXPECT_EQ(lcore.deleteProduct(id), result);
}

INSTANTIATE_TEST_CASE_P(
        CoreTestProductParametrizedTest,
        CoreTestProductParametrized,
        ::testing::Values(
                ProductTestStruct{0, Core::MethodResult::SUCCESS}
                ProductTestStruct{9755765, Core::MethodResult::SUCCESS},
                ProductTestStruct{2147483647, Core::MethodResult::SUCCESS}
        ));
```
Nie daje gwarancji, że to się skompiluje, bo pisałem z pamięci